projects/dashboards/src/test/dashboard-verification.ts

```diff
@@ -22,11 +23,20 @@ export const isValidModelJson = (
 
 export const mockDashboardProviders = [
   mockProvider(GraphQlRequestService),
-  mockProvider(TimeRangeService),
+  mockProvider(ColorService),
+  mockProvider(TimeRangeService, {
+    getTimeRangeAndChanges: () => EMPTY
+  }),
   mockProvider(NavigationService, {
     getAllValuesForQueryParameter: () => []
   }),
   mockProvider(MetadataService, {
     getFilterAttributes: () => of([])
-  })
+  }),
+  mockProvider(LoggerService, {
+    warn: jest.fn().mockImplementation(fail),
```

Comment:
Where is this `fail` coming from?

Reply:
jest - all of its functions are ambient ( i.e. test, describe, expect)